docusaurus/docs/React/custom-code-examples/livestream-setup.mdx

```diff
@@ -0,0 +1,77 @@
+
+```tsx
+<VirtualizedMessageList stickToBottomScrollBehavior="auto" />
+```
+
+## Specify the default item height
+
+The virtualized message list supports items with varying heights and automatically keeps track of the rendered item sizes, while estimating the not-yet-rendered sizes based on latest message during load. To optimize this further and minimize potential recalculations, set the `defaultItemHeight` to the height of your usual one-line message (If in doubt, use the developer tools inspector to determine the sizing). 
+
+<img src={MessageHeight} alt='Message Height' width='700' />
+
+The element above is `58px` tall, so let's put that at the `defaultItemHeight` prop value:
+
+```tsx
+<VirtualizedMessageList defaultItemHeight={58} />
+```
+
+## Avoid vertical CSS margins
+
+This could be a potential pitfall if you're using a custom message component. The virtualized message list uses `getBoundingClientRect` ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect)) on the outermost message element. This method does not measure CSS margins, which are tricky to measure in the first place [due to collapsing](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing)). If the custom message has vertical margins that affect its height, the message list might behave erratically (blinking, rendering white areas, etc). 
+
+Use your DOM inspector and look for any orange gaps between the message elements. See the screenshot below for example of headings with margins:
+
+<img src={MarginInspector} alt='Margin Inspector' width='700' />
+
+## Choose the right Giphy size
+
+This practice is not unique to the virtualized message list but might help you build a more compact chat timeline and reduce the overall traffic for your users. By default, the Giphy attachments [use the original Giphy version image](https://developers.giphy.com/docs/api/schema#image-object). You can override that by setting the `giphyVersion` property of the `Channel` Component to a smaller one - use the keys [from the Giphy documentation](https://developers.giphy.com/docs/api/schema#image-object).
+
+```tsx
+<Chat client={client}>
+  <ChannelList />
+  <Channel giphyVersion="fixed_height_small">
+    <VirtualizedMessageList />
+    <MessageInput />
+  </Channel>
+</Chat>
+```
+
+## Design reactions so that they don't change the message size
+
+If you're using a custom message, make sure that adding reactions to the message does not increase the message height. Good examples for such design would be the chat boxes of Twitch and Telegram. 
```

Comment:
One final best practice: make sure that adding reactions to the message does not increase the message height if you're using a custom `Message` component. Good examples for such design would be the chat boxes of Twitch and Telegram.